sensors/errors_cron_sensor.py

```diff
@@ -0,0 +1,291 @@
+
+        self._logger.info("Current problem rules: {0}".format(self.kv_enforcements))
+
+        for rule in rules:
+            cron_time = CronTab(self.convert_to_crontab(rule.trigger['parameters']))
+
+            # Gets the previous run time in seconds
+            previous_cron_s = cron_time.previous(now=utc_date_time, delta=False)
+            next_cron_s = cron_time.next(now=utc_date_time, delta=False)
+
+            # converts from seconds to a useable datetime format
+            previous_cron_stamp = datetime.datetime.utcfromtimestamp(previous_cron_s)
+            next_cron_stamp = datetime.datetime.utcfromtimestamp(next_cron_s)
+
+            # Adds the timezone information to the time stamp
+            previous_cron_dt = previous_cron_stamp.replace(tzinfo=pytz.UTC)
+            next_cron_dt = next_cron_stamp.replace(tzinfo=pytz.UTC)
+
+            enforcements = self.st2_client.ruleenforcements.query(rule_ref=rule.ref)
+            if len(enforcements) > 0:
+                self._logger.info("Checking enforcements for rule: {0}".format(rule.ref))
+                result = self.check_enforcements(enforcements, previous_cron_dt, next_cron_dt)
+
+                if result:
+                    self.delete_from_kv(rule.ref)
+            else:
+                st2_comments = "Cron job is not running and no enforcements can be found"
+                self.dispatch_trigger(st2_rule_name=rule.ref,
+                                      st2_server=self.st2_fqdn,
+                                      st2_comments=st2_comments)
+
+        self._sensor_service.set_value(name=self.kv_sensor_name, value=self.kv_enforcements)
+
+    def check_enforcements(self, enforcements, previous_cron, next_cron):
+        """ Checks all the enforments to find if the cron was executed and to
+        get the status of the execution if one can be found
+        """
+        for enforcement in enforcements:
+            # Parse the enforcement enforced time to be in datetime format for comparison
+            # We need to drop the microseconds from the datetime format for the comparison
+            # to be accurate
+            enforced_at = parse(enforcement.enforced_at).replace(microsecond=0)
+
+            # Creating a buffer of one run time
+            cron_delta = next_cron - previous_cron
+            cron_low_buffer = previous_cron - cron_delta
+
+            if cron_low_buffer <= enforced_at <= next_cron:
+                # If the rule logic fails then no execution_id exists in the rule enforcement
+                if hasattr(enforcement, 'execution_id'):
+                    st2_execution = self.st2_client.liveactions.get_by_id(enforcement.execution_id)
+                    if st2_execution.status in PROGRESS_STATUSES:
+                        self._logger.info("Currently running execution. Will check next run")
+                        return False
+                    elif st2_execution.status in ERRORED_STATUSES:
```

Comment:
If its in a `ERRORED_STATUSES` state lets try to use the common error methods to find an error and return it with the trigger.